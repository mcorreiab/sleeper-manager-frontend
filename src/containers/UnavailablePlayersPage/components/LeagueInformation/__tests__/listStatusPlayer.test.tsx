import renderer from "react-test-renderer";
import ListStatusPlayer from "../listStatusPlayer";

const players = [
  {
    id: "1",
    name: "Player 1",
    position: "Position 1",
    team: "Team 1",
  },
  {
    id: "2",
    name: "Player 2",
    position: "Position 2",
    team: "Team 2",
  },
];

it("given that is not the last status, should render the list with success", () => {
  const tree = renderer
    .create(
      <ListStatusPlayer
        lastStatus={false}
        statusLabel="Status 1"
        players={players}
      />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it("given that is the last status, should render the list with success", () => {
  const tree = renderer
    .create(
      <ListStatusPlayer
        lastStatus={true}
        statusLabel="Status 1"
        players={players}
      />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
