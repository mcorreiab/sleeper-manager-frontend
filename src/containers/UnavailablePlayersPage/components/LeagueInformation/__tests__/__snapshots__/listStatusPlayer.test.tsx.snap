// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`given that is not the last status, should render the list with success 1`] = `
<div
  className="flex flex-col"
>
  <h3
    className="font-bold text-[#dbdfff] rounded-[32px] bg-sm-blue self-start text-xs py-1 px-2 mb-[0.625rem]"
  >
    Status 1
  </h3>
  <ul
    className="list-none text-sm text-sm-lightgray flex flex-col items-start"
  >
    <li
      className="grid grid-cols-4-autofr gap-y-2 w-full lg:py-4 lg:pl-6 lg:bg-sm-dark-blue lg:rounded-lg"
    >
      <p
        className="col-start-1 row-start-1 col-span-4 text-sm-lightwhite"
      >
        Player 1
      </p>
      <p
        aria-label="Player 1 position"
        className="col-start-1 row-start-2 text-sm-lightgray text-sm"
      >
        Position 1
      </p>
      <svg
        aria-hidden="true"
        className="my-0 mx-1 col-start-2 row-start-2 self-center"
        fill="none"
        height="4"
        viewBox="0 0 4 4"
        width="4"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="2"
          cy="2"
          fill="#8F97AB"
          r="2"
        />
      </svg>
      <p
        aria-label="Player 1 NFL team"
        className="col-start-3 row-start-2 text-sm-lightgray text-sm"
      >
        Team 1
      </p>
    </li>
    <li
      className="w-full my-2"
    >
      <hr
        className="border border-t-0 border-[#ffffff26] border-solid"
      />
    </li>
    <li
      className="grid grid-cols-4-autofr gap-y-2 w-full lg:py-4 lg:pl-6 lg:bg-sm-dark-blue lg:rounded-lg"
    >
      <p
        className="col-start-1 row-start-1 col-span-4 text-sm-lightwhite"
      >
        Player 2
      </p>
      <p
        aria-label="Player 2 position"
        className="col-start-1 row-start-2 text-sm-lightgray text-sm"
      >
        Position 2
      </p>
      <svg
        aria-hidden="true"
        className="my-0 mx-1 col-start-2 row-start-2 self-center"
        fill="none"
        height="4"
        viewBox="0 0 4 4"
        width="4"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="2"
          cy="2"
          fill="#8F97AB"
          r="2"
        />
      </svg>
      <p
        aria-label="Player 2 NFL team"
        className="col-start-3 row-start-2 text-sm-lightgray text-sm"
      >
        Team 2
      </p>
    </li>
  </ul>
  <hr
    className="border border-t-0 border-[#ffffff26] border-solid mt-2 mb-6 mx-0"
  />
</div>
`;

exports[`given that is the last status, should render the list with success 1`] = `
<div
  className="flex flex-col"
>
  <h3
    className="font-bold text-[#dbdfff] rounded-[32px] bg-sm-blue self-start text-xs py-1 px-2 mb-[0.625rem]"
  >
    Status 1
  </h3>
  <ul
    className="list-none text-sm text-sm-lightgray flex flex-col items-start"
  >
    <li
      className="grid grid-cols-4-autofr gap-y-2 w-full lg:py-4 lg:pl-6 lg:bg-sm-dark-blue lg:rounded-lg"
    >
      <p
        className="col-start-1 row-start-1 col-span-4 text-sm-lightwhite"
      >
        Player 1
      </p>
      <p
        aria-label="Player 1 position"
        className="col-start-1 row-start-2 text-sm-lightgray text-sm"
      >
        Position 1
      </p>
      <svg
        aria-hidden="true"
        className="my-0 mx-1 col-start-2 row-start-2 self-center"
        fill="none"
        height="4"
        viewBox="0 0 4 4"
        width="4"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="2"
          cy="2"
          fill="#8F97AB"
          r="2"
        />
      </svg>
      <p
        aria-label="Player 1 NFL team"
        className="col-start-3 row-start-2 text-sm-lightgray text-sm"
      >
        Team 1
      </p>
    </li>
    <li
      className="w-full my-2"
    >
      <hr
        className="border border-t-0 border-[#ffffff26] border-solid"
      />
    </li>
    <li
      className="grid grid-cols-4-autofr gap-y-2 w-full lg:py-4 lg:pl-6 lg:bg-sm-dark-blue lg:rounded-lg"
    >
      <p
        className="col-start-1 row-start-1 col-span-4 text-sm-lightwhite"
      >
        Player 2
      </p>
      <p
        aria-label="Player 2 position"
        className="col-start-1 row-start-2 text-sm-lightgray text-sm"
      >
        Position 2
      </p>
      <svg
        aria-hidden="true"
        className="my-0 mx-1 col-start-2 row-start-2 self-center"
        fill="none"
        height="4"
        viewBox="0 0 4 4"
        width="4"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="2"
          cy="2"
          fill="#8F97AB"
          r="2"
        />
      </svg>
      <p
        aria-label="Player 2 NFL team"
        className="col-start-3 row-start-2 text-sm-lightgray text-sm"
      >
        Team 2
      </p>
    </li>
  </ul>
</div>
`;
